app/assets/javascripts/components/projects/ProjectBoard.js

```diff
@@ -21,134 +28,150 @@ import SearchResults from './../search/SearchResults';
+    const { sprintIndex: sprintDragIndex, columnId: dragColumn } = JSON.parse(source.droppableId);
+    const { index: sourceIndex } = source;
+    const { index: destinationIndex} = destination;
+    const isSameColumn = dragColumn === dropColumn;
+    const destinationArray = getArray(dropColumn, newBacklogSprints, newChillyBinStories, sprintDropIndex); // stories of destination column
+    const sourceArray = getArray(dragColumn, newBacklogSprints, newChillyBinStories, sprintDragIndex); // stories of source column
+    const dragStory = sourceArray[sourceIndex];
+
+    if (!destination) return;
+    if (isSameColumn && sourceIndex === destinationIndex) return;
+
+    const newPosition = getNewPosition(
+      destinationIndex,
+      sourceIndex,
+      destinationArray,
+      isSameColumn,
+      dragStory.storyType,
+    );
+
+    const newStories = moveTask(
+      sourceArray,
+      destinationArray,
+      sourceIndex,
+      destinationIndex,
+    );
+
+    // Changing the column array order
+    if (dropColumn === columns.CHILLY_BIN) {
+      setNewChillyBinStories(newStories);
     }
-  ];
-
-  const sideBarButtons = [
-    {
-      description: I18n.t('revert_columns_tooltip'),
-      onClick: reverseColumns,
-      'data-id': 'reverse-button',
-      isVisible: projectBoard.reverse,
-      icon: 'fas fa-columns'
-    },
-    {
-      description: I18n.t('toggle_column', { column: 'chilly bin' }),
-      onClick: () => toggleColumn('chillyBin'),
-      'data-id': 'toggle-chilly-bin',
-      isVisible: projectBoard.visibleColumns.chillyBin,
-      icon: 'fas fa-snowflake'
-    },
-    {
-      description: I18n.t('toggle_column', { column: 'backlog' }),
-      onClick: () => toggleColumn('backlog'),
-      'data-id': 'toggle-backlog',
-      isVisible: projectBoard.visibleColumns.backlog,
-      icon: 'fas fa-th-list'
-    },
-    {
-      description: I18n.t('toggle_column', { column: 'done' }),
-      onClick: () => toggleColumn('done'),
-      'data-id': 'toggle-done',
-      isVisible: projectBoard.visibleColumns.done,
-      icon: 'fas fa-check-circle'
+
+    if (dropColumn === columns.BACKLOG) {
+      setNewBacklogSprints(getNewSprints(newStories, newBacklogSprints, sprintDropIndex));
     }
-  ];
 
-    const columnProps = {
-      'data-id': projectBoard.reverse ? 'reversed-column' : 'normal-column',
-      columns: projectBoard.reverse ? columns.reverse() : columns
-    };
+    // Persisting the new array order
+    const newState = getNewState(isSameColumn, dropColumn, dragStory.state);
+    return dragDropStory(dragStory.id, dragStory.projectId, {
+      position: newPosition,
+      state: newState,
+    });
+  };
+
+  const onDragUpdate = ({ source, destination }) => {
+    dragStory(source, destination, newBacklogSprints, 
+      newBacklogSprintsUpdated => setNewBacklogSprints(newBacklogSprintsUpdated)
+    );
+  };
```

Comment:
You don't need to declare a new function
```js
newBacklogSprintsUpdated => setNewBacklogSprints(newBacklogSprintsUpdated)
    );
```
since `setNewBacklogSprints` is already a function 😄 

```diff
const onDragUpdate = ({ source, destination }) => {
  dragStory(source, destination, newBacklogSprints, 
-   newBacklogSprintsUpdated => setNewBacklogSprints(newBacklogSprintsUpdated)
+   setNewBacklogSprints
  );
};
```

Reply:
Thank you nolleto! I forgot this.